src/modules/handlers/ManifestHandler.js

```diff
@@ -19,40 +19,88 @@ DB.version(1).stores({
 });
 
 function SetNextManifestCheck() { localStorage.setItem('nextManifestCheck', new Date().getTime() + (1000 * 60 * 60)); }
-export function StoreManifest(callback) {
-  callback({ status: 'storingManifest', statusText: 'Storing Manifest...', error: false, loading: true, manifestMounted: false });
-  Promise.all([
-    DB.table('DestinyActivityDefinition').toCollection().first(),
-    DB.table('DestinyActivityTypeDefinition').toCollection().first(),
-    DB.table('DestinyActivityModeDefinition').toCollection().first(),
-    DB.table('DestinyCollectibleDefinition').toCollection().first(),
-    DB.table('DestinyPresentationNodeDefinition').toCollection().first(),
-    DB.table('DestinyRecordDefinition').toCollection().first(),
-    DB.table('DestinyInventoryItemLiteDefinition').toCollection().first(),
-    DB.table('DestinyObjectiveDefinition').toCollection().first(),
-    DB.table('DestinyProgressionDefinition').toCollection().first(),
-    DB.table('DestinyTalentGridDefinition').toCollection().first(),
-    DB.table('DestinyVendorDefinition').toCollection().first()
-  ]).then(async (values) => {
+
+const manifestLoadAttempts = 0;
+const maxLoadAttempts = 3;
+export async function StoreManifest(callback) {
+  try {
+    if (manifestLoadAttempts > maxLoadAttempts) {
+      throw `Manifest is corrupted after ${maxLoadAttempts} attempts.`;
+    }
+    manifestLoadAttempts++;
+    callback({
+      status: "storingManifest",
+      statusText: "Storing Manifest...",
+      error: false,
+      loading: true,
+      manifestMounted: false
+    });
+    let isManifestRefreshRequired = false;
+
+    let values = await Promise.all([
+      DB.table("DestinyActivityDefinition").toCollection().first(),
+      DB.table("DestinyActivityTypeDefinition").toCollection().first(),
+      DB.table("DestinyActivityModeDefinition").toCollection().first(),
+      DB.table("DestinyCollectibleDefinition").toCollection().first(),
+      DB.table("DestinyPresentationNodeDefinition").toCollection().first(),
+      DB.table("DestinyRecordDefinition").toCollection().first(),
+      DB.table("DestinyInventoryItemLiteDefinition").toCollection().first(),
+      DB.table("DestinyObjectiveDefinition").toCollection().first(),
+      DB.table("DestinyProgressionDefinition").toCollection().first(),
+      DB.table("DestinyTalentGridDefinition").toCollection().first(),
+      DB.table("DestinyVendorDefinition").toCollection().first()
+    ]);
+
     //Add data to global manifest object
     MANIFEST = {
-      "DestinyActivityDefinition": values[0].data,
```

Comment:
These fail due to `values[#]` not having a data prop since they failed to grab. So this can be easy fixed by adding `values[#]?.data` which would either set the data or undefined, which the next check should pick up?